skimage/feature/sift.py

```diff
@@ -0,0 +1,730 @@
+            radius_patch = math.sqrt(2) * radius
+            p_min = np.asarray(
+                np.maximum(0, center_pos - radius_patch[:, np.newaxis] + 0.5),
+                dtype=int)
+            p_max = np.asarray(
+                np.minimum(center_pos + radius_patch[:, np.newaxis] + 0.5,
+                           (dim[0] - 1, dim[1] - 1)), dtype=int)
+
+            for k in range(len(p_max)):
+                rad_k = radius[k]
+                ori = orientations[k]
+                histograms = np.zeros((self.n_hist, self.n_hist, self.n_ori),
+                                      dtype=float_dtype)
+                # the patch
+                r, c = np.meshgrid(np.arange(p_min[k, 0], p_max[k, 0]),
+                                   np.arange(p_min[k, 1], p_max[k, 1]),
+                                   indexing='ij', sparse=True)
+                # normalized coordinates
+                r_norm = np.subtract(r, center_pos[k, 0], dtype=float_dtype)
+                c_norm = np.subtract(c, center_pos[k, 1], dtype=float_dtype)
+                r_norm, c_norm = self._rotate(r_norm, c_norm, ori)
+
+                # select coordinates and gradient values within the patch
+                inside = np.maximum(np.abs(r_norm), np.abs(c_norm)) < rad_k
+                r_norm, c_norm = r_norm[inside], c_norm[inside]
+                r_idx, c_idx = np.nonzero(inside)
+                r = r[r_idx, 0]
+                c = c[0, c_idx]
+                gradient_row = gradient[0][r, c, scales[k]]
+                gradient_col = gradient[1][r, c, scales[k]]
+                # compute the (relative) gradient orientation
+                theta = np.arctan2(gradient_col, gradient_row) - ori
+                lam_sig = self.lambda_descr * sigma[k]
+                # Gaussian weighted kernel magnitude
+                kernel = np.exp((r_norm * r_norm + c_norm * c_norm)
+                                / (-2 * lam_sig ** 2))
+                magnitude = np.sqrt(gradient_row * gradient_row
+                                    + gradient_col * gradient_col) * kernel
+
+                lam_sig_ratio = 2 * lam_sig / self.n_hist
+                rc_bins = (hists - (1 + self.n_hist) / 2) * lam_sig_ratio
+                rc_bin_spacing = lam_sig_ratio
+                ori_bins = (2 * np.pi * bins) / self.n_ori
+
+                # distances to the histograms and bins
+                dist_r = np.abs(np.subtract.outer(rc_bins, r_norm))
+                dist_c = np.abs(np.subtract.outer(rc_bins, c_norm))
+
+                # the orientation histograms/bins that get the contribution
+                near_t, near_t_val = _ori_distances(ori_bins, theta)
+
+                # create the histogram
+                _update_histogram(histograms, near_t, near_t_val, magnitude,
+                                  dist_r, dist_c, rc_bin_spacing)
+
+                # convert the histograms to a 1d descriptor
+                histograms = histograms.flatten()
```

Comment:
We can avoid a copy here :wink:
```suggestion
                histograms = histograms.reshape(-1)
```